### DISCUSSION TITLE
Contributing to this repo

### DISCUSSION THREAD

**Discussion by neelchakrabrty** [archlinux/archinstall#1802] [Q&A] [answered]
Hello, so far I have been testing the script and reporting bugs as I find them,I enjoy doing this a lot thanks to users like [svartkanin](https://github.com/svartkanin) and [Torxed](https://github.com/Torxed). 

I would like to help by writing code too, but I don't know much python to begin with. What should be one's background in python to understand the code written in this project, and what's a good place to start studying the source code? Should I start with `/archinstall/__init__.py`?

I apologize if this is not the right place to ask this.

**Comment by Torxed** [comment, marked as answer]
This would be the right place to ask, so you're on the right track.

In order of difficulty – starting with the easiest – I would say:

 * Profiles (like the i3 profile etc)
 * Installer code (installer.py)
 * Disk code
 * Menu code (this might be easier than the disk code, but they're both tricky)

Im'd say the best place to start regardless is in a place of interest or need – and ask for help :)
